packages/backend-core/src/Base.ts

```diff
@@ -263,7 +266,9 @@ export class Base {
     }
 
     // In cross-origin requests the use of Authorization header is mandatory
-    if (isCrossOrigin) {
+    // TODO: The origin header can be present in same-origin POST, PUT, PATCH requests etc.
```

Comment:
@colinclerk @SokratisVidros 
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin#description 

It seems that we need to do the full check as `checkCrossOriginReferrer` to address the API endpoint use case.
1. Frontend requests a same-origin API resource from the server
2. Authorization is in the cookie and not authorization header.
3. Origin might be included as stated in the source.
4. We would return 401 (_?_)

Reply:
So is it ok if we keep the `checkCrossOriginReferrer` there for now still ?
_(if we do, we will change the name to `checkCrossOrigin`)_

Reply:
yes